MapboxCoreNavigation/RouteController.swift

```diff
@@ -122,30 +138,77 @@ open class RouteController: NSObject {
      */
     public var snapsUserLocationAnnotationToRoute = true
     
+    var isRerouting = false
     var lastRerouteLocation: CLLocation?
     
     var routeTask: URLSessionDataTask?
     
+    /// :nodoc: This is used internally when the navigation UI is being used
+    public var usesDefaultUserInterface = false
+    
+    var sessionState = SessionState()
+    var outstandingFeedbackEvents = [CoreFeedbackEvent]()
+    
     /**
      Intializes a new `RouteController`.
      
      - parameter route: The route to follow.
      - parameter directions: The Directions object that created `route`.
      - parameter locationManager: The associated location manager.
+     - parameter accessToken: Optional Mapbox accessToken.
      */
-    @objc(initWithRoute:directions:locationManager:)
-    public init(along route: Route, directions: Directions = Directions.shared, locationManager: NavigationLocationManager = NavigationLocationManager()) {
+    @objc(initWithRoute:directions:locationManager:accessToken:)
+    public init(along route: Route, directions: Directions = Directions.shared, locationManager: NavigationLocationManager = NavigationLocationManager(), accessToken: String? = nil) {
         self.directions = directions
         self.routeProgress = RouteProgress(route: route)
         self.locationManager = locationManager
         self.locationManager.activityType = route.routeOptions.activityType
         super.init()
         
         self.locationManager.delegate = self
+        self.resumeNotifications()
+        self.startEvents(accessToken: accessToken)
+        self.resetSession()
     }
     
     deinit {
         suspendLocationUpdates()
+        checkAndSendOutstandingFeedbackEvents(forceAll: true)
+        sendCancelEvent()
+        suspendNotifications()
+    }
+    
+    func startEvents(accessToken: String? = nil) {
+        var mapboxAccessToken: String? = nil
+        if let accessToken = accessToken {
+            mapboxAccessToken = accessToken
+        } else if let path = Bundle.main.path(forResource: "Info", ofType: "plist"),
+            let dict = NSDictionary(contentsOfFile: path) as? [String: AnyObject],
+            let token = dict["MGLMapboxAccessToken"] as? String {
+            mapboxAccessToken = token
+        }
+        
+        if let mapboxAccessToken = mapboxAccessToken {
+            events.isDebugLoggingEnabled = true
+            events.isMetricsEnabledInSimulator = true
```

Comment:
I don't think we should ship this with metrics enabled in simulator set to true. Maybe this library can define a new user default key (it would not need to be documented) that could be checked and this value could default to false and and use the value in user defaults if available?